### PR TITLE
Add bytes or sbytes to enum where available 

### LINKED ISSUE #973: The great Enum Space Waste of 2020  (aka why are enums not byte sized)
This is probably of importance to networking in particular:
Enums default to `int` sized. It is highly likely that *every single enum in the codebase* should be `byte` sized for network savings.
Using up a byte of space is better than using 4 and depending on the compressor to handle the 3 extra zero bytes.

### PR DESCRIPTION
**Pull request by Ygg01**
Fixes #973 

Sets all enums to be `byte` sized (or `sbyte` where possible).

There were a few exceptions that didn't allow YamlSerializer to pass.

### CODE REVIEW

**Review by gradientvera — APPROVED** [review]
This *seems* fine but maybe someone else should take a look at it first.

**Comment by PaulRitter** [comment]
inb4 packet size drops by 500%

**Comment by Ygg01** [comment]
Yeah. I wanted to mark his as draft. I suspect space-station-14 needs to be changed too, right?

**Comment by gradientvera** [comment]
> Yeah. I wanted to mark his as draft. I suspect space-station-14 needs to be changed too, right?

You mean content? The tests seem to past on content so I'd say it's fine.

File: Robust.Client/Graphics/Clyde/Clyde.Constants.cs [hunk 1]
```diff
@@ -35,7 +35,7 @@ internal sealed partial class Clyde
         // To be clear: You shouldn't change this. This just helps with understanding where Primitive Restart is being used.
         private const ushort PrimitiveRestartIndex = ushort.MaxValue;
 
-        private enum Renderer
+        private enum Renderer : short
```
Comment (exp111) [inline Robust.Client/Graphics/Clyde/Clyde.Constants.cs:38]: Can't you make this sbyte?

Comment (Ygg01) [reply Robust.Client/Graphics/Clyde/Clyde.Constants.cs:38]: Think I can. Really should have set this PR to draft

**Comment by Ygg01** [comment]
> inb4 packet size drops by 500%

If the server doesn't work, then the packet size is 0. Infinite% reduction!

**Comment by Ygg01** [comment]
@Zumorica can we test the Robust-Toolbox local version against SS14? I want to see that Serializer isn't crapping on my parade.

**Comment by gradientvera** [comment]
> @Zumorica can we test the Robust-Toolbox local version against SS14? I want to see that Serializer isn't crapping on my parade.

That's what the "Test content master against engine / build" test does

**Comment by Ygg01** [comment]
I should have worded that more precisely.  I want to start SS14 project using local RobustToolbox version. Even if temporarily. Tests are ok, but I don't have too great of confidence in them,

**Comment by gradientvera** [comment]
Just check your branch in the RobustToolbox submodule locally then.

**Comment by SaphireLattice** [comment]
If this is done for serialisation purposes, then shouldn't that happen within serializer and setting it to use bytes for enums? This is unlikely to help the actual processing as the memory words are still fair bit larger than a byte.

**Comment by metalgearsloth** [comment]
> 
> 
> If this is done for serialisation purposes, then shouldn't that happen within serializer and setting it to use bytes for enums? This is unlikely to help the actual processing as the memory words are still fair bit larger than a byte.

Isn't it to help with what's being sent over the wire?

**Comment by PJB3005** [comment]
> If this is done for serialisation purposes, then shouldn't that happen within serializer and setting it to use bytes for enums? This is unlikely to help the actual processing as the memory words are still fair bit larger than a byte.

This mostly changes the size of the enum on the heap. NetSerializer uses variable length int writing (that I do not know a formal name of) so a value like `(int) 5` still only gets written as one byte. The only place this really helps is `byte` enums that actually manage to use the last bit (which isn't any from what I can tell).

**Review by PJB3005 — APPROVED** [review]

**Pull request merged**